google/cloud/aiplatform/models.py

```diff
@@ -2389,3 +2438,624 @@ def export_model(
+
+                Overrides encryption_spec_key_name set in aiplatform.init.
+            staging_bucket (str):
+                Optional. Bucket to stage local model artifacts. Overrides
+                staging_bucket set in aiplatform.init.
+        Returns:
+            model: Instantiated representation of the uploaded model resource.
+        Raises:
+            ValueError: If only `explanation_metadata` or `explanation_parameters`
+                is specified.
+                Also if model directory does not contain a supported model file.
+        """
+        # https://cloud.google.com/vertex-ai/docs/predictions/pre-built-containers#xgboost
+        XGBOOST_SUPPORTED_VERSIONS = ["0.82", "0.90", "1.1", "1.2", "1.3", "1.4"]
+        XGBOOST_CONTAINER_IMAGE_URI_TEMPLATE = (
+            "{registry}/vertex-ai/prediction/xgboost-{cpu_or_gpu}.{version}:latest"
+        )
+
+        XGBOOST_SUPPORTED_MODEL_FILE_EXTENSIONS = [
+            ".pkl",
+            ".joblib",
+            ".bst",
+        ]
+
+        _LOGGER.info(f"Using the {xgboost_version} version of XGBoost.")
+
+        if xgboost_version not in XGBOOST_SUPPORTED_VERSIONS:
+            _LOGGER.warning(
+                f"XGBoost version {xgboost_version} is not supported. "
+                f"Supported versions: {XGBOOST_SUPPORTED_VERSIONS}"
+            )
+
+        container_image_uri = XGBOOST_CONTAINER_IMAGE_URI_TEMPLATE.format(
+            registry=_get_container_registry(
+                location or aiplatform.initializer.global_config.location
+            ),
+            cpu_or_gpu="cpu",
+            version=xgboost_version.replace(".", "-"),
+        )
+
+        model_file_path_obj = pathlib.Path(model_file_path)
+        if not model_file_path_obj.is_file():
+            raise ValueError(
+                f"model_file_path path must point to a file: '{model_file_path}'"
+            )
+
+        model_file_extension = model_file_path_obj.suffix
+        if model_file_extension not in XGBOOST_SUPPORTED_MODEL_FILE_EXTENSIONS:
+            _LOGGER.warning(
+                f"Only the following XGBoost model file extensions are currently supported: '{XGBOOST_SUPPORTED_MODEL_FILE_EXTENSIONS}'"
+            )
+            _LOGGER.warning(
+                "Treating the model file as a binary serialized XGBoost Booster."
+            )
+            model_file_extension = ".bst"
+
+        # Preparing model directory
+        # We cannot clean up the directory immediately after calling Model.upload since
+        # that call may be asynchronous and return before the model file has been read.
+        # The temporary data will be automatically cleaned up by the system later.
+        prepared_model_dir = tempfile.mkdtemp()
+        prepared_model_file_path = pathlib.Path(prepared_model_dir) / (
+            "model" + model_file_extension
+        )
+        shutil.copy(model_file_path_obj, prepared_model_file_path)
+
+        return aiplatform.Model.upload(
```

Comment:
I can do that, but that somewhat limits the ability of the derived classes to override the `.upload` function with different signature, since we switching direct call with "virtual" call.

Reply:
Done.